src/OpenTelemetry/Internal/CircularBuffer.cs

```diff
@@ -151,6 +155,8 @@ public bool TryAdd(T value, int maxSpinCount)
                         return false; // exceeded maximum spin count
                     }
 
+                    spinWait.SpinOnce();
```

Comment:
if we use spinWait, then it keeps track of the count - so we can eliminate our own calculation (spinCountDown)

Reply:
Thanks, reworked it